### PR TITLE
replace yapf with black

### PR DESCRIPTION
**Pull request by asottile**
here's the alternate proposal, use `black` instead of `yapf`

___

### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>

  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?

</details>

### CODE REVIEW

**Comment by asottile** [comment]
this one also finishes much quicker at ~35s

File: kornia/augmentation/base.py [hunk 1]
```diff
@@ -341,28 +343,29 @@ def inverse_transform(
...
-            list,
-            tuple,
-        )):
+        if isinstance(
+            input,
+            (
+                list,
+                tuple,
+            ),
+        ):
```
Comment (Borda) [inline kornia/augmentation/base.py:368]: this is formatting I dont like about Black :D

**Review by Borda — COMMENTED** [review]
I do not mind about this Blac, just to note if you run YAPF again after Black it will be the best one :P

File: kornia/augmentation/base.py (hunk 1 above)
Comment (carmocca) [reply kornia/augmentation/base.py:368]: This is because the line has a "magic comma" aka "trailing comma" so black expands it.

Ideally, black would evaluate whether the expression can be written in a single line given the maximum line length (even if it has a trailing comma) and then it would choose whether to expand it or not.
But I don't think black does that.

@asottile First of all, you are a legend, big fan of your work!
I was wondering if https://github.com/asottile/add-trailing-comma could be expanded to always remove trailing commas.

This way we could run your tool, and then black - thus making sure that all trialing commas are there because the expression wouldn't fit otherwise

Comment (asottile) [reply kornia/augmentation/base.py:368]: so `add-trailing-comma` already removes unnecessary inline commas -- so maybe it just needs to be run once before black?

Comment (asottile) [reply kornia/augmentation/base.py:368]: p.s. I also don't really personally like black, but often it's the "easiest" choice

Comment (carmocca) [reply kornia/augmentation/base.py:368]: > so add-trailing-comma already removes unnecessary inline commas -- so maybe it just needs to be run once before black?

Only does it if in a single line:

```diff
-if isinstance(input, (list, tuple,),):
+if isinstance(input, (list, tuple)):
     pass
```

But using the case above, this conversion does not happen:
```diff
 if isinstance(
     input,
     (
        list,
-       tuple,
+       tuple
-    ),
+    )
 ):
     pass
```

If it did, we could then run `black` and get what we want:

``` python
if isinstance(input, (list, tuple)):
    pass
```

Since the `isinstance(...)` check is well under the line length limit if on one line.

So what I would like is a `remove-trailing-comma` tool that can be run before `black`
This way black is forced to re-evaluate whether to expand expressions or not.

Do you prefer if we move this discussion elsewhere?

Comment (Borda) [reply kornia/augmentation/base.py:368]: @carmocca I found reverse https://github.com/asottile/add-trailing-comma

Comment (asottile) [reply kornia/augmentation/base.py:368]: @carmocca yeah -- what I'm suggesting is to put `add-trailing-comma` before `black` inside the `.pre-commit-config.yaml` -- this would ensure that inline extra-commas are removed before black messes them up

Comment (asottile) [reply kornia/augmentation/base.py:368]: even better -- there's a `--skip-magic-trailing-comma` option to `black` -- ezpz!

let me recreate this PR using that option

Comment (asottile) [reply kornia/augmentation/base.py]: done!

File: kornia/feature/laf.py [hunk 2]
```diff
@@ -494,12 +488,12 @@ def extract_patches_from_pyramid(
             if (scale_mask.float().sum()) == 0:
                 continue
             scale_mask = (scale_mask > 0).view(-1)
-            grid = generate_patch_grid_from_normalized_LAF(cur_img[i:i + 1], nlaf[i:i + 1, scale_mask, :, :], PS)
+            grid = generate_patch_grid_from_normalized_LAF(cur_img[i : i + 1], nlaf[i : i + 1, scale_mask, :, :], PS)
```
Comment (Borda) [inline kornia/feature/laf.py:491]: is this fine with Flake8?

**Review by Borda — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -15,21 +15,18 @@ repos:
...
-        name: formatting
-        entry: yapf
-        args: [--parallel]
-        language: python
-        types: [python]
+      - id: black
+        args: [
+            --line-length=120,
+            --skip-string-normalization,
+            --skip-magic-trailing-comma,
```
Comment (Borda) [inline .pre-commit-config.yaml:25]: can we add this to some config file so anyone who would run black separately gets the same result?

File: kornia/feature/laf.py (hunk 2 above)
Comment (asottile) [reply kornia/feature/laf.py:491]: yep, that's the `E203` -- I made sure to follow the readme that black provides

(by the way, I maintain flake8)

File: .pre-commit-config.yaml (hunk 3 above)
Comment (asottile) [reply .pre-commit-config.yaml:25]: yep I can put this into a `pyproject.toml` (the only configuration that black supports)

File: kornia/feature/siftdesc.py [hunk 4]
```diff
@@ -77,12 +77,25 @@ class SIFTDescriptor(nn.Module):
     """
 
     def __repr__(self) -> str:
-        return self.__class__.__name__ +\
-            '(' + 'num_ang_bins=' + str(self.num_ang_bins) +\
-            ', ' + 'num_spatial_bins=' + str(self.num_spatial_bins) +\
-            ', ' + 'patch_size=' + str(self.patch_size) +\
-            ', ' + 'rootsift=' + str(self.rootsift) +\
-            ', ' + 'clipval=' + str(self.clipval) + ')'
+        return (
```
Comment (edgarriba) [inline kornia/feature/siftdesc.py:80]: @asottile is there any way to not destroy that much this ones ? @carmocca was suggesting to use f-strings, which possibly means revisiting big part of the library

Comment (asottile) [reply kornia/feature/siftdesc.py:80]: yeah I would probably move those to f-strings -- though you might be able to get away with a little helper that's something like this:

```python
return ''.join((
    f'{type(self).__name__}(',
    ', '.join(f'{name}={getattr(self, name)}' for name in sorted(self.__dict__) if not name.startswith('_')),
    ')'
))
```

(warning: written in github ui, might need some tweaking -- but the basic idea is to pull the attributes off of the object instead of having to hardcode all of them)

Comment (edgarriba) [reply kornia/feature/siftdesc.py:80]: cool , thanks. We'll try to address that  /cc @shijianjian

Comment (edgarriba) [reply kornia/feature/siftdesc.py:80]: https://github.com/kornia/kornia/issues/1053

**Review by edgarriba — APPROVED** [review]

**Comment by edgarriba** [comment]
@asottile are we good to merge ?

**Comment by edgarriba** [comment]
@asottile thanks so much for this amazing contribution. Feel free to join us in our slack :-)
https://join.slack.com/t/kornia/shared_invite/zt-csobk21g-CnydWe5fmvkcktIeRFGCEQ
